Docs: Badger file permission as non-root service (#5282)
jaegertracing/jaeger: plugin/storage/badger/docs/storage-file-non-root-permission.md

> # Badger file permissions as non-root service

Reviewer: there should be a link to this file from somewhere.

Author: Yes, I will open PR at `jaegertracing
/
documentation` after this PR merged.

Like https://github.com/jaegertracing/documentation/pull/685

> After the release of 1.50, Jaeger's Docker image is no longer running with root privileges (in [#4783](https://github.com/jaegertracing/jaeger/pull/4783)), and issues such as "permission denied" occur when writing.

Reviewer: ```suggestion
After the release of 1.50, Jaeger's Docker image is no longer running with root privileges (in [#4783](https://github.com/jaegertracing/jaeger/pull/4783)). In some installations it may cause issues such as "permission denied" errors when writing data.
```

> If you encounter this type of problem, you can update the owner of the data file before starting Jaeger.
> 
> Here is a sample `docker-compose.yml`:

Reviewer: ```suggestion
A possible workaround for this ([proposed here](https://github.com/jaegertracing/jaeger/issues/4906#issuecomment-1991779425)) is to run an initialization step as `root` that pre-creates the Badger data directory and updates its owner to the user that will run the main Jaeger process.
```

```yaml
version: "3.9"

services:
[...]
  jaeger:
    image: jaegertracing/all-in-one:latest
    command:
      - "--badger.ephemeral=false"
      - "--badger.directory-key=/badger/data/keys"
      - "--badger.directory-value=/badger/data/values"
      - "--badger.span-store-ttl=72h0m0s" # limit storage to 72hrs
    environment:
      - SPAN_STORAGE_TYPE=badger

>     volumes:

Reviewer: ```suggestion
    # Mount host directory "jaeger_badger_data" as "/badger" inside the container.
    # The actual data directory will be "/badger/data", 
    # since we cannot change permissions on the mount.
    volumes:
```

      - jaeger_badger_data:/badger
    ports:
      - "16686:16686"
      - "14250"
      - "4317"
    depends_on:

>       init-jaeger:

Reviewer: ```suggestion
      prepare-data-dir:
```

        condition: service_completed_successfully

>   init-jaeger:

Reviewer: ```suggestion
  prepare-data-dir:
```

>     user: root

Reviewer: ```suggestion
    # Run this step as root so that we can change the directory owner.
    user: root
```

    image: jaegertracing/all-in-one:latest
    command: "/bin/sh -c 'mkdir -p /badger/data && touch /badger/data/.initialized && chown -R 10001:10001 /badger/data'"
    volumes:
      - jaeger_badger_data:/badger

volumes:
  jaeger_badger_data:
```

> ## Reference
> 
> - [jaeger#4906](https://github.com/jaegertracing/jaeger/issues/4906#issuecomment-1991779425)

Reviewer: ```suggestion
```
